File: LeetCode_Solutions/LeetCode_Solutions/SameTree_100.m
//
//  SameTree_100.m
//  LeetCode_Solutions
//
//  Created by 来自遥远星系的核心巡洋舰 on 16/10/13.
//  Copyright © 2016年 KnightJoker. All rights reserved.
//



#import "SameTree_100.h"

@implementation SameTree_100

- (BOOL)isSameTree:(struct TreeNode *)p and:(struct TreeNode *)q {
    
    if ( p == nil && q == nil) {
        return YES;
    } else if (p == nil || q == nil) {
        return NO;
    }
    
    if (p->val != q->val) {
        return NO;
    } else
        return [self isSameTree:p->left and:q->left] && [self isSameTree:p->right and:q->right];

}

@end

File: LeetCode_Swift/LeetCode_Swift/SameTree_100.swift
//
//  SameTree_ 100.swift
//  LeetCode_Swift
//
//  Created by Huni on 10/08/2017.
//  Copyright © 2017 KnighhtJoker. All rights reserved.
//

import Foundation

public class SameTree_100 {
    
    func isSameTree(_ p: TreeNode?, _ q: TreeNode?) -> Bool {
        if p == nil && q == nil {
            return true
        } else if p == nil || q == nil {
            return false
        }
        
        if p?.val != q?.val {
            return false
        } else {
            return self.isSameTree(p?.left, q?.left) && self.isSameTree(p?.right, q?.right)
        }
        
    }
    
}

